lib/config/config-ops.js

```diff
@@ -268,5 +272,84 @@ module.exports = {
      */
     isEverySeverityValid(config) {
         return Object.keys(config).every(ruleId => this.isValidSeverity(config[ruleId]));
+    },
+
+    /**
+     * Merges all configurations in a given config vector. A vector is an array of config file paths, each optionally
+     * followed by one or more numbers that correspond to the indices of nested config blocks within the config file's
+     * overrides section. All config data is assumed to be cached.
+     * @param {Array<number|string>} vector array of config file paths or relative override indices
+     * @returns {Object} config object
+     */
+    getConfigFromVector(vector) {
+
+        const subvector = Array.from(vector);
+        let config,
+            subConfigs,
+            nearestCacheIndex = vector.length - 1;
+
+        // Check the merged vector config cache to try to find a merged config for the current config or a parent
+        for (; nearestCacheIndex >= 0; nearestCacheIndex--) {
+            config = configCache.getMergedVectorConfig(subvector);
+            if (config) {
+                break;
+            }
+            subvector.pop();
+        }
+
+        if (config) {
```

Comment:
This is not a very helpful comment, but all of the code that goes below seems like a black magic to me. Branching on array value types, hardcoded array indexes, etc. I sort of understand what the code is doing, but it's not very readable and is really hard to follow. Not sure what can be done to improve this (as I said, not a very helpful comment).